presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py

```diff
@@ -23,25 +27,62 @@ def __init__(self, analyzer_engine: AnalyzerEngine = None, ocr: OCR = None):
             ocr = TesseractOCR()
         self.ocr = ocr
 
-    def analyze(self, image: object, **kwargs) -> List[ImageRecognizerResult]:
+    def analyze(
+        self, image: object, ocr_kwargs: Optional[dict] = None, **text_analyzer_kwargs
+    ) -> List[ImageRecognizerResult]:
         """Analyse method to analyse the given image.
 
-        :param image: PIL Image/numpy array or file path(str) to be processed
-        :param kwargs: Additional values for the analyze method in AnalyzerEngine
+        :param image: PIL Image/numpy array or file path(str) to be processed.
+        :param ocr_kwargs: Additional params for OCR methods.
+        :param text_analyzer_kwargs: Additional values for the analyze method
+        in AnalyzerEngine.
 
-        :return: list of the extract entities with image bounding boxes
+        :return: List of the extract entities with image bounding boxes.
         """
-        ocr_result = self.ocr.perform_ocr(image)
+        # Perform OCR
+        perform_ocr_kwargs, ocr_threshold = self._parse_ocr_kwargs(ocr_kwargs)
+        ocr_result = self.ocr.perform_ocr(image, **perform_ocr_kwargs)
+
+        # Apply OCR confidence threshold if it is passed in
+        if ocr_threshold is not None:
+            ocr_result = self.threshold_ocr_result(ocr_result, ocr_threshold)
+
+        # Analyze text
         text = self.ocr.get_text_from_ocr_dict(ocr_result)
 
         analyzer_result = self.analyzer_engine.analyze(
-            text=text, language="en", **kwargs
+            text=text, language="en", **text_analyzer_kwargs
         )
         bboxes = self.map_analyzer_results_to_bounding_boxes(
             analyzer_result, ocr_result, text
         )
         return bboxes
 
+    @staticmethod
+    def threshold_ocr_result(ocr_result: dict, ocr_threshold: float) -> dict:
+        """Filter out OCR results below confidence threshold.
+
+        :param ocr_result: OCR results (raw).
+        :param ocr_threshold: Threshold value between -1 and 100.
+
+        :return: OCR results with low confidence items removed.
+        """
+        if ocr_threshold < -1 or ocr_threshold > 100:
+            raise ValueError("ocr_threshold must be between -1 and 100")
+
+        # Get indices of items above threshold
+        idx = list()
+        for i, val in enumerate(ocr_result["conf"]):
+            if float(val) >= ocr_threshold:
+                idx.append(i)
+
+        # Only retain high confidence items
+        filtered_ocr_result = {}
+        for key in list(ocr_result.keys()):
+            filtered_ocr_result[key] = [ocr_result[key][i] for i in idx]
+
+        return filtered_ocr_result
```

Comment:
nit: Because of the not-so-straightforward structure we get the results in, would be much easier to query the confidence while mapping into bounding boxes (`map_analyzer_results_to_bounding_boxes`). When iterating ocr results - keep processing into BBs only if above threshold.

Reply:
That is a good point, however, `map_analyzer_results_to_bounding_boxes` takes in `text` as an argument. I placed the ocr thresholding inside `analyze` before `get_text_from_ocr_dict` is called such that we only have to filter out in one location rather than two (e.g., `ocr_results` and `text`).

Reply:
got it, makes sense